exercises/tuples-lists-dictionaries/challenge_072.py

```diff
@@ -5,10 +5,47 @@
 Delete the subject they have chosen from the list before you display the list again.
 """
 
-subjects_list = ["Maths", "Literature", "Science", "History", "Spanish", "Computing"]
-print(subjects_list)
-dislike = input("Which of these subjects do you dislike? ")
-erase_from_list = subjects_list.index(dislike)
-del subjects_list[erase_from_list]
-print(subjects_list)
+subjects_list = [
+    "Maths",
+    "Literature",
+    "Science",
+    "History",
+    "Spanish",
+    "Computing",
+    "Physics",
+    "Geometry",
+    "Geography",
+    "Economy",
+    "Italian",
+    "Physical Education",
+]
+subjects_list_lowercase = [subject.lower() for subject in subjects_list]
 
+
+def join_subjects(subjects_list):
+    return ", ".join(subjects_list)
+
+
+print(f"Subjects list: {join_subjects(subjects_list)}")
+
+
+def ask_user_input():
+    while True:
+        user_input = input("Which one of these subjects do you dislike? ")
+        user_input_lowercase = user_input.lower()
+
+        if user_input_lowercase in subjects_list_lowercase:
+            index_to_delete = subjects_list_lowercase.index(user_input_lowercase)
+            subject_to_delete = subjects_list[index_to_delete]
+
+            del subjects_list[index_to_delete]
```

Comment:
Can you think of a way to do this without the double delete action (one of which must be redundant).

Reply:
I have removed the deletion of the lowercase list items.